### DISCUSSION TITLE
async computed field on return block.

### DISCUSSION THREAD

**Discussion by henriquetroiano** [prisma/prisma#16732] [Q&A]
## Problem

I have a tickets database.
Then I need to get each ticket usage.
This is a promise.
This is a fast thing, once I'm using mongodb. But keeps a promise anyway.
Need to async/await inside compute block.
Make a loop on each one of the results is not the best thing, once this can be done automatically with this feature as async.

## Suggested solution

![image](https://user-images.githubusercontent.com/63327237/206866838-39a0bbe4-5c1f-4e2d-8873-2198746ff465.png)
(it's just an usage example)

## Alternatives

No alternatives on my mind.

## Additional context

nothing extra

**Comment by nurul3101** [comment]
Hey @henriquetroiano 👋 

It looks like you are trying to perform a query inside compute, does this [example](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result#add-a-custom-method-to-the-result-object) work as a reference for you? It achieves a similar operation as a query is performed inside compute block, and it is awaited when the custom function is invoked.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
